doc/rm/c_cpp_coding_style.md

```diff
@@ -257,6 +257,36 @@ The code using an X Macro must:
     It is usually clearer if you split the definition over multiple lines.
 *   Immediately after the definition, include the `.def` file for the X Macro.
 
+### Undefined Behavior
+
+***All uses of Undefined Behavior (unless stated otherwise) are forbidden without explicit approval of toolchain experts.***
+
+As in any high-quality C/C++ project, Undefined Behavior (as defined by C++14 or C11 as the case may be) must be treated as a security risk.
+To have reasonable hope of being able to upgrade compilers without nasty surprises, we code strictly against the standards.
+
+Although not comprehensive, the following popular idioms are actually Undefined Behavior as defined by the standards.
+This list is provided to illustrate the subtle nature of UB, and how it runs counter to the semantics of C many programmers may have acquired over time.
+
+*   Casting an arbitrary `void *` to `uint32_t *` and dereferencing it.
+    This is the most common accidental violation of the Strict Aliasing Rule.
+*   Creating unaligned pointers.
+    C and C++ consider merely creating such pointers, not just dereferencing them, to be UB.
+*   Comparing (by equality or order) non-null pointers of distinct provenance (e.g. pointers into different stack variables).
+*   Using vanilla `memcpy()` on `volatile` memory; any non-`volatile` access to `volatile` memory is UB.
+
+However, because this is a firmware project, sometimes C and C++ cannot express precisely what we need to do, usually due to oversights in how the languages were originally specified.
+Intentionally invoking undefined behavior is a measure of last resort that should be signed off on by the project's toolchain experts.
+Most of the time, there is a sanctioned (but non-obvious) way to perform a particular operation.
+Justification for invocation of UB should include desired assembly output patterns and reasons why writing inline assembly is insufficient (such as inappropriate generating of optimization barriers).
+
+For example, `memcpy()` can be used to work around Strict Aliasing without incurring a performance penalty (as one might expect from the fact that it's a function call).
+
+Plenary dispensation is given to the following operations, which are UB according to the standards but which no sensible compiler treats as such.
```

Comment:
`Plenary dispensation` is unusual language. Can you make it a bit plainer?

Maybe something as simple as `The following uses of undefined behavior have been assessed as low risk and may be used as required throughout the project.`